### PR TITLE
GUI Sprint 1

### PR DESCRIPTION
**Pull request by Drayz64**

### CODE REVIEW

File: src/main/java/tempest/Module.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 public class Module {
     private UUID id;
     private String name;
-    private LinkedList<StudySession> studySessions = new LinkedList<StudySession>();
+    private LinkedList<StudySession> studySessions = new LinkedList<>();
```
Comment (bridges-wood) [inline src/main/java/tempest/Module.java:19]: Why remove the type annotation?

File: src/main/java/tempest/State.java [hunk 2]
```diff
@@ -5,14 +5,14 @@
 import java.util.List;
 
 public class State {
-  private List<Module> modules = new ArrayList<Module>();
+  private static List<Module> modules = new ArrayList<>();
```
Comment (bridges-wood) [inline src/main/java/tempest/State.java:8]: Why remove the type annotation here as well?

File: src/main/java/tempest/Module.java (hunk 1 above)
Comment (Drayz64) [reply src/main/java/tempest/Module.java:19]: IntelliJ doesn't like them, says they are unnecessary: "Explicit type argument StudySession can be replace with <>"

File: src/test/java/tempest/ModuleViewTest.java [hunk 3]
```diff
@@ -57,7 +69,7 @@ public void testAddingASession() {
                 break;
             }
         }
-
+        int previousSessions = testModule.getStudySessions().length;
```
Comment (bridges-wood) [inline src/test/java/tempest/ModuleViewTest.java:72]: This last test fails to add sessions.

File: src/main/java/tempest/Module.java (hunk 1 above)
Comment (bridges-wood) [reply src/main/java/tempest/Module.java:19]: Okay, sweet.

File: src/test/java/tempest/ModuleViewTest.java (hunk 3 above)
Comment (Drayz64) [reply src/test/java/tempest/ModuleViewTest.java:72]: In what way? The test doesn't do as described or the test fails when you run it?

Comment (bridges-wood) [reply src/test/java/tempest/ModuleViewTest.java:72]: The test fails.
image

Comment (Drayz64) [reply src/test/java/tempest/ModuleViewTest.java:72]: Strange its fine for me:

Test Running

**Review by Drayz64 — COMMENTED** [review]
Maybe these are helpful?

https://stackoverflow.com/questions/5218870/getting-a-headlessexception-no-x11-display-variable-was-set

https://stackoverflow.com/questions/5362512/unable-to-run-java-gui-programs-with-ubuntu

File: src/test/java/tempest/ModuleViewTest.java (hunk 3 above)
Comment (bridges-wood) [reply src/test/java/tempest/ModuleViewTest.java:72]: Okay, that is weird

**Comment by bridges-wood** [comment]
> Maybe these are helpful?
> 
> https://stackoverflow.com/questions/5218870/getting-a-headlessexception-no-x11-display-variable-was-set
> 
> https://stackoverflow.com/questions/5362512/unable-to-run-java-gui-programs-with-ubuntu

I think if we somehow create a virtual display buffer it might work?

**Pull request merged**